File: node/object-listar.js
#!/usr/bin/env node

const { S3Client, ListObjectsV2Command } = require("@aws-sdk/client-s3");
const { NodeHttpHandler } = require("@aws-sdk/node-http-handler");
const fs = require('fs');
const { Command } = require('commander');
const program = new Command();

const credentials = JSON.parse(fs.readFileSync('credentials.json'));

const s3Client = new S3Client({
    endpoint: credentials.endpoint_url,
    region: "us-east-1",
    credentials: {
        accessKeyId: credentials.access_key,
        secretAccessKey: credentials.secret_key
    },
    requestHandler: new NodeHttpHandler({
        httpsAgent: new require('https').Agent({
            rejectUnauthorized: false
        })
    }),
    forcePathStyle: true  // Força o estilo de caminho para o endpoint
});

program
    .requiredOption('--bucket-name <bucketName>', 'Nome do bucket que contém os objetos');

program.parse(process.argv);

const options = program.opts();

const listObjects = async () => {
    try {
        const data = await s3Client.send(new ListObjectsV2Command({ Bucket: options.bucketName }));
        if (data.Contents.length === 0) {
            console.log(`Nenhum objeto encontrado no bucket ${options.bucketName}.`);
        } else {
            data.Contents.forEach(obj => {
                const objUrl = `${credentials.endpoint_url}/${options.bucketName}/${obj.Key}`;
                console.log(`${obj.Key} - ${objUrl}`);
            });
        }
    } catch (err) {
        console.error('Erro ao listar objetos:', err);
    }
};

listObjects();

File: c#/object-listar.cs
using System;
using Amazon.S3;
using Amazon.S3.Model;
using Newtonsoft.Json.Linq;
using System.IO;

class ListObjects
{
    static async Task Main(string[] args)
    {
        if (args.Length != 2 || args[0] != "--bucket-name")
        {
            Console.WriteLine("Usage: dotnet run --bucket-name <bucket-name>");
            return;
        }

        string bucketName = args[1];
        var credentials = JObject.Parse(File.ReadAllText("credentials.json"));

        var config = new AmazonS3Config
        {
            ServiceURL = credentials["endpoint_url"].ToString(),
            ForcePathStyle = true
        };

        var s3Client = new AmazonS3Client(credentials["access_key"].ToString(), credentials["secret_key"].ToString(), config);

        try
        {
            var listObjectsRequest = new ListObjectsV2Request
            {
                BucketName = bucketName
            };

            var listObjectsResponse = await s3Client.ListObjectsV2Async(listObjectsRequest);

            if (listObjectsResponse.S3Objects.Count == 0)
            {
                Console.WriteLine($"Nenhum objeto encontrado no bucket {bucketName}.");
            }
            else
            {
                foreach (var obj in listObjectsResponse.S3Objects)
                {
                    string objectUrl = $"{credentials["endpoint_url"]}/{bucketName}/{obj.Key}";
                    Console.WriteLine($"{obj.Key} - {objectUrl}");
                }
            }
        }
        catch (AmazonS3Exception e)
        {
            Console.WriteLine($"Erro ao listar objetos: {e.Message}");
        }
    }
}
